src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/BindingInfo.cs

```diff
@@ -100,46 +100,45 @@ public static BindingInfo GetBindingInfo(IEnumerable<object> attributes)
                 }
             }
 
-            // PropertyBindingPredicateProvider
-            var predicateProviders = attributes.OfType<IPropertyBindingPredicateProvider>().ToArray();
-            if (predicateProviders.Length > 0)
+            // PropertyFilterProvider
+            var propertyFilterProviders = attributes.OfType<IPropertyFilterProvider>().ToArray();
+            if (propertyFilterProviders.Length > 0)
             {
                 isBindingInfoPresent = true;
-                bindingInfo.PropertyBindingPredicateProvider = new CompositePredicateProvider(
-                    predicateProviders);
+                bindingInfo.PropertyFilterProvider = new CompositePropertyFilterProvider(propertyFilterProviders);
             }
 
             return isBindingInfoPresent ? bindingInfo : null;
         }
 
-        private class CompositePredicateProvider : IPropertyBindingPredicateProvider
+        private class CompositePropertyFilterProvider : IPropertyFilterProvider
```

Comment:
I'm not sure whether loop unrolling is necessary here and suggest the existing single closure might be sufficiently efficient. Further the existing code is pretty readable.

I do suggest a minor optimization for a likely-common case:
``` c#
if (propertyFilterProviders.Length == 1)
{
    bindingInfo.PropertyFilterProvider = propertyFilterProviders[0];
}
else if (propertyFilterProviders.Length > 1)
{
    bindingInfo.PropertyFilterProvider = new CompositePropertyFilterProvider(propertyFilterProviders);
}
```

If we then find `CompositePropertyFilterProvider` instances are expensive, we can always chain later.